templates/CRM/Contribute/Form/Contribution/Confirm.tpl

```diff
@@ -111,10 +111,30 @@
             {else}
               {if $frequency_interval > 1}
                 <p>
-                  <strong>{ts 1=$frequency_interval 2=$frequency_unit}I want to contribute this amount every %1 %2s.{/ts}</strong>
+                  {if $frequency_unit eq 'day'}
+                    <p><strong>{ts 1=$frequency_interval}I want to contribute this amount every every %1 days.{/ts}</strong></p>
+                  {elseif $frequency_unit eq 'week'}
+                    <p><strong>{ts 1=$frequency_interval}I want to contribute this amount every every %1 weeks.{/ts}</strong></p>
+                  {elseif $frequency_unit eq 'month'}
+                    <p><strong>{ts 1=$frequency_interval}I want to contribute this amount every every %1 months.{/ts}</strong></p>
+                  {elseif $frequency_unit eq 'year'}
+                    <p><strong>{ts 1=$frequency_interval}I want to contribute this amount every every %1 years.{/ts}</strong></p>
+                  {else}
```

Comment:
I think we can avoid the final "else"? #19655 does not have one.

Reply:
Yes, I know, but I put it in as a precaution, so it works even if someone changes it or another unit is introduced in the future 😉